plugins/payment-plugin/payment_plugin/payment.py

```diff
@@ -0,0 +1,181 @@
+import base64
+import json
+import requests
+import uuid
+from enum import Enum
+from typing import Any, Dict, Optional
+from jb_manager_bot import AbstractFSM
+from jb_manager_bot.abstract_fsm import Variables
+from jb_manager_bot.data_models import (
+    FSMOutput,
+    Status,
+    FSMIntent,
+    Message,
+    MessageType,
+    TextMessage,
+)
+
+
+def generate_auth_header(client_id: str, client_secret: str) -> str:
+    auth_string = base64.b64encode(bytes(f"{client_id}:{client_secret}", "utf-8")).decode("utf-8")
+    return f"Basic {auth_string}"
+
+def request_payment(amount: int, client_id: str, client_secret: str, reference_id: str) -> str:
+    url = "https://api.razorpay.com/v1/payment_links/"
+
+    payload = json.dumps({"amount": amount * 100, "currency": "INR", "reference_id": reference_id})
+    headers = {
+        "Content-type": "application/json",
+        "Authorization": generate_auth_header(client_id=client_id, client_secret=client_secret),
+    }
+
+    response = requests.request("POST", url, headers=headers, data=payload)
+    response = json.loads(response.text)
+
+    payment_url = response["short_url"]
+    return payment_url
+
+
+class PaymentPluginReturnStatus(Enum):
+    SUCCESS = "success"
```

Comment:
I think the ENUM values need to be capital letters to be compatible with the YAML

Reply:
@sameersegal codegen should access enum by name, not by value. In that case, value does not matter. 
For convention, we can have same string for names and values.